projects/storefrontlib/src/cms-structure/seo/structured-data/builders/product/product-schema.builder.ts

```diff
@@ -0,0 +1,54 @@
+import { Inject, Injectable, Optional } from '@angular/core';
+import { Product } from '@spartacus/core';
+import { combineLatest, Observable, of } from 'rxjs';
+import { first, flatMap, map } from 'rxjs/operators';
+import { CurrentProductService } from '../../../../../cms-components/product/current-product.service';
+import { JsonLdBuilder, SchemaBuilder } from '../schema.interface';
+import { JSONLD_PRODUCT_BUILDER } from '../tokens';
+
+/**
+ * Adds the minimal structured data for the product, see https://schema.org/product.
+ * The actual data collection is delegated to `JsonLdBuilder`s, which can be injected
+ * using the `JSONLD_PRODUCT_BUILDER` token.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductSchemaBuilder implements SchemaBuilder {
+  constructor(
+    private currentProduct: CurrentProductService,
+    @Optional()
+    @Inject(JSONLD_PRODUCT_BUILDER)
+    protected builders: JsonLdBuilder<Product>[]
+  ) {}
+
+  build(): Observable<any> {
+    return this.currentProduct.getProduct().pipe(
+      // tslint:disable-next-line:deprecation
+      // startWith(null),
+      flatMap((product: Product) => combineLatest(this.collect(product))),
+      first(Boolean),
+      map((res: {}[]) => {
```

Comment:
Could be probably simplified to one liner `map((res: {}[]) => Object.assign({}, ...res)),`